### ISSUE TITLE
[Suggestions] Suggest new Idea for Cloud Function Demos 💡

### ISSUE THREAD

**Issue by Glide7** [appwrite/demos-for-functions#29]
## Suggest new Idea for Cloud Function Demos 💡
We use Appwrite Cloud Functions demos as a good way to show how Cloud Functions can be used with Appwrite, and also as a good starting point for developers wanting to write their new functions.

We would love to receive your suggestions for other cloud function demos!
Please comment down below if you have such an idea ♥

### Before starting any work ⚙
* Please wait for Appwrite team feedback 💙
* Let us know if you wish to start a new issue and be assigned to work on a function demo
* A separate issue will be opened once an idea is approved  
* What language you want to help with book 🧾

**Comment by mlucas-NU** [comment]
Here are some ideas to get the conversation started. Ultimately, they're meant to target granular questions someone might ask, so some of these would need to be combined depending on the format of the demo. Come to think of it, what type of demos are you imagining?

- The simplest cronjob, like logging "hello world."
- The simplest endpoint, like returning the current server time.
- Basic cronjob/endpoint examples that interact with appwrite's services.
- An explanation of endpoint security. E.g. can we create endpoints that can be accessed without authentication? Can we restrict endpoints to certain users/teams?
- Endpoints with required/optional parameters.
- Endpoints with popular response types (e.g. plaintext, JSON, XML, CSV).
- Custom wrappers for REST functions. For example, add an `expiration` timestamp field 5 days in the future on document PUTs, do not include `expiration` in document GET responses, do not allow users to modify `expiration` through any tricky means, and do not allow users to interact with the document at all after the `expiration` date has passed.
- An example of interacting with an external API that appwrite users are likely to use (e.g. cloud hosting, social media). It would be nice to use an API that has its own beginner-friendly explanation (e.g. Google Adwords' basic API](https://developers.google.com/adwords/api/docs/guides/first-api-call)) in case the user wants to understand that API as well, but there's a risk that the explanation changes over time or is moved/deleted. In particular, I'd be interested to know how to securely store API keys in appwrite (this may be a challenge without superadmin accounts) and see a code snippet that retrieves these keys, authenticates with the external API, and does a thing.
- Code Function best practices. Can appwrite containers ingest a directory of Code Functions as a volume when deployed? This would be incredibly useful for source control and continuous deployment/scaling. If so, how should the directory be structured? How does a developer specify to appwrite that a function is an endpoint or a cronjob? Are there any other recommendations for teams making maintainable Cloud Functions?

**Comment by mendoza** [comment]
@Glide7 What do you think about
- Server heartbeat (same as server time on @mlucas-NU)
- Getting weather on a specific Geolocation with coordinates or even with a city name!

**Comment by lexxyungcarter** [comment]
- targetting specific collections when their documents get added/updated
- A direct comparison/mapping of "here's how you did it in firebase" > "here's how you do it in appwrite" (not sure how  best to word that)
